src/reducers/player.js

```diff
@@ -1,35 +1,48 @@
-import {ActionTypes} from '../constants';
-const {INITIAL_STATE, TILE_SELECTED} = ActionTypes;
+import {ActionTypes, CellStates} from '../constants';
+const {INITIAL_STATE, RACK_TILE_SELECTED, END_TURN, BOARD_TILE_SELECTED} = ActionTypes;
 
 export default function player(state = {}, action) {
-  if (action.turn !== state.id && action.type !== INITIAL_STATE) {
+  if (action.player !== state.id || (action.turn !== state.id && action.type !== INITIAL_STATE)) {
     return state;
   }
 
   switch (action.type) {
-    case TILE_SELECTED:
-      if (state.selected == '') {
+    case RACK_TILE_SELECTED:
+      if (state.selected < 0) {
         return Object.assign({}, state, {
-          selected: action.tileId
+          selected: action.selected
         });
       } else {
-        var newletter, oldletter, newletters;
-        newletter = state.letters[parseInt(action.tileId.substring(2, 3))];
-        oldletter = state.letters[parseInt(state.selected.substring(2, 3))];
-        newletters = state.letters.slice();
-        newletters[parseInt(state.selected.substring(2, 3))] = newletter;
-        newletters[parseInt(action.tileId.substring(2, 3))] = oldletter;
+        let newletters = state.letters.slice();
+        let oldletter = newletters[state.selected];
+        newletters[state.selected] = newletters[action.selected];
+        newletters[action.selected] = oldletter;
```

Comment:
i'm not sure if i agree with the functionality here.  consider the following case: 

1. the user has a tile rack with [a, b, c]
2. the user selects `b` by mistake.  he meant to select `a`.
3. the user clicks on `a` to select it.  
4. instead, the tile moves.

this functionality is okay for this pull request.  but, you may want to consider separating `MOVE` and `SELECT` functionality.